From 4c5fd5b2044c8d8d51aa00beca3bc4965c518c50 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 01:51:05 -0400
Subject: [iommu] vt-d: Remove to_pci_dev() in intel_map_page()

Message-id: <20140905015105.31297.78168.stgit@gir.stowe>
Patchwork-id: 91346
O-Subject: [RHEL7.1 PATCH 53/65] iommu/vt-d: Remove to_pci_dev() in intel_map_page()
Bugzilla: 1129880 1087643
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1087643
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1129880
Upstream Status: 46333e375f3228196e8cb741d817165f9793b46c

commit 46333e375f3228196e8cb741d817165f9793b46c
Author: David Woodhouse <David.Woodhouse@intel.com>
Date:   Mon Mar 10 20:01:21 2014 -0700

    iommu/vt-d: Remove to_pci_dev() in intel_map_page()

    It might not be...

    Signed-off-by: David Woodhouse <David.Woodhouse@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/intel-iommu.c b/drivers/iommu/intel-iommu.c
index 37ce54b..0a8e166 100644
--- a/drivers/iommu/intel-iommu.c
+++ b/drivers/iommu/intel-iommu.c
@@ -3023,7 +3023,7 @@ static dma_addr_t intel_map_page(struct device *dev, struct page *page,
      struct dma_attrs *attrs)
 {
  return __intel_map_single(dev, page_to_phys(page) + offset, size,
-      dir, to_pci_dev(dev)->dma_mask);
+      dir, *dev->dma_mask);
 }
 
 static void flush_unmaps(void)
-- 
1.7.1